demos/asp-net-core/ITOps.ViewModelComposition/EventHandler.cs

```diff
@@ -4,8 +4,5 @@
 
 namespace ITOps.ViewModelComposition
 {
-    interface ISubscription
-    {
-        Task Invoke(dynamic viewModel, object @event, RouteData routeData, IQueryCollection query);
-    }
+    public delegate Task EventHandler(dynamic pageViewModel, dynamic @event, RouteData routeData, IQueryCollection query);
```

Comment:
Subscribers now receives a `dynamic` event, where before with zero effort for the subscriber and one single `internal` interface and an `internal` class in the engine is was strongly typed and much better for subscribers

Reply:
Fair enough, this was my one concern with this approach, but I wasn't sure how much it mattered, since the subscribers already have to be "dynamic-aware" wrt to the view model. However, if the event can be typed, then it's true that we should probably do that. I'll restore the type safety.